do_kafka_test.sh

```diff
@@ -1,71 +1,172 @@
 #!/bin/bash
-./start_kafka.sh
 
-# Wait until Kafka creates our topics.
-while true; do
-    ntopics=$(docker exec jpocvdp_kafka_1 /opt/kafka/bin/kafka-topics.sh --list --zookeeper 172.17.0.1 | wc -l)
+# This script tests the PPM against a kafka cluster. It sets up variables for container and input data
+# file names. It starts a Kafka container using another script and checks that required topics are created.
+# If the container or topics are missing, the script exits. It builds a Docker image using the current
+# directory and specified name/tag. It runs a series of tests using a script with different properties
+# and input data files, outputting results to the console. It stops the Kafka container after the tests
+# are completed. The script performs five steps: set up the test environment, wait for Kafka to create
+# topics, build the PPM image, run the tests, and clean up.
 
-    if [[ $ntopics == "4" ]]; then 
-        echo 'Found 4 topics:'
-        docker exec jpocvdp_kafka_1 /opt/kafka/bin/kafka-topics.sh --list --zookeeper 172.17.0.1 2> /dev/null
-        
-        break   
-    fi
-
-    sleep 1
-done
+CYAN='\033[0;36m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
 
+CURRENT_DIR_NAME=${PWD##*/}
+KAFKA_CONTAINER_NAME=$CURRENT_DIR_NAME-kafka-1
 MAP_FILE=data/I_80.edges
 BSM_DATA_FILE=data/I_80_test.json
 TIM_DATA_FILE=data/I_80_test_TIMS.json
+PPM_CONTAINER_NAME=test_ppm_instance
+PPM_IMAGE_TAG=do-kafka-test-ppm-image
+PPM_IMAGE_NAME=jpo-cvdp_ppm
+
+setup() {
+    if [ -z $DOCKER_HOST_IP ]; then
+        echo "DOCKER_HOST_IP is not set. Exiting."
+        exit 1
+    fi
+
+    # print setup info
+    echo "=== Setup Info ==="
+    echo "DOCKER_HOST_IP: $DOCKER_HOST_IP"
+    echo "KAFKA_CONTAINER_NAME: $KAFKA_CONTAINER_NAME"
```

Comment:
![image](https://github.com/usdot-jpo-ode/jpo-cvdp/assets/62157949/bf8d7aa2-95ee-43af-bc8a-f3baadd1294f)
The kafka container name is separated by underscore not hyphen.

Reply:
On my machine, the kafka container name is separated by a hyphen. This may have to do with differences in Docker Desktop versions. I am using v4.17.0, what are you using?

We may want to consider modifying the script to check for container names separated by an underscore and a hyphen, just to cover all of our bases.